Move modules unit tests into modules and clean

loganharbour opened the pull request:
Closes #16568 and closes #16571.

Will hopefully resolve #16092 in here in a bit.

loganharbour committed:
Move modules specific unit tests into the respective modules

refs #16568

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/651534/) on e3ff119 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/16570/site)

This comment will be updated on new commits.

loganharbour committed:
Use proper comparisons in water fluid props test

refs #16568

loganharbour committed:
Rename ray tracing unit tests for consistency

refs #16568

loganharbour committed:
Use TEST_F for GeochemicalSystem unit tests

refs #16571

This resolves an issue on clang in which the common member
variables within GeochemicalSystemTest were crashing
upon construction

loganharbour commented:
@WilkAndy take a look at 46f15a8. It resolves the issue discussed in #16571.

loganharbour committed:
Resolve comparison warnings in geochemistry unit tests

refs #16568

loganharbour committed:
Resolve comparison warnings for porous flow unit tests

refs #16568

loganharbour committed:
Resolve comparison warnings for stochastic tools unit tests

refs #16568

loganharbour commented:
The framework unit tests in `unit` no longer build any modules. The module-specific unit tests have been moved into their respective unit directories.

Decided to not include #16092 in this because we're still not sure what's going on there.

It may be worth discussing the movement of `unit` to `framework/unit`.
